### DISCUSSION TITLE
New ARIA roles for (code) changes

### DISCUSSION THREAD

**Discussion by Malvoz** [community/community#14030] [Accessibility] [answered]
The ARIA 1.3 draft defines new roles appropriate for code changes in pull requests:

- [`suggestion`](https://w3c.github.io/aria/#suggestion)
- [`insertion`](https://w3c.github.io/aria/#insertion)
- [`deletion`](https://w3c.github.io/aria/#deletion)

Applying these roles as appropriate to code changes could improve the experience for screen reader users (in supporting UAs) when reviewing code.

**Comment by benjiallen** [comment]
@Malvoz very interesting. Thanks for making me aware of these roles. I see that `insertion` and `deletion` are already in the 1.2 ARIA spec. I'm wondering why I've not heard of these roles before. Do you know if assistive technology support is documented anywhere? Do you know of a service that is using these roles successfully?

Given I've not heard of these roles, my hunch is that they are not well supported. I hope I'm wrong!

**Reply by benjiallen** [reply]
@Malvoz any thoughts on my questions?

**Reply by Malvoz** [reply]
> I see that insertion and deletion are already in the 1.2 ARIA spec. I'm wondering why I've not heard of these roles before

Ah yes, I suppose I mentioned the 1.3 draft because it also includes the `suggestion` role ([example](https://w3c.github.io/aria/#example-16)) which may be suitable specifically for when [suggesting changes](https://docs.github.com/assets/cb-20237/images/help/pull_requests/commit-suggestion-button.png).

> Do you know of a service that is using these roles successfully?

I do not. NVDA supports the `<ins>` and `<del>` elements ([current status](https://github.com/nvaccess/nvda/issues/10789)) but perhaps they do not yet support the respective ARIA roles. I may file an issue this week, or just outright test it.

And for browsers, a quick search shows that at least Chrome supports the [`insertion` and `deletion`](https://bugs.chromium.org/p/chromium/issues/detail?id=945366) roles, [as well as `suggestion`](https://chromestatus.com/feature/4666935918723072).

> Do you know if assistive technology support is documented anywhere?

I haven't found any support data other than the Chrome bugs, though it should be available at MDN or [a11ysupport.io](https://a11ysupport.io/) sooner or later.

**Reply by benjiallen** [reply]
Thanks for the information, I'll review as we continue our research on this problem.

**Comment by cehfisher** [comment, marked as answer]
Hello @Malvoz

Thank you for your feedback, and we genuinely appreciate your input. Our team has reviewed your request and we want to be transparent about our priorities. This particular request is not currently being actively worked on. This doesn’t imply that we’re not considering it in the future, but for now, it is not on our development roadmap.

We encourage you to check our [public roadmap](https://github.com/github/roadmap/projects/1) to get insights into the projects we are currently focusing on for the upcoming year. Please note that smaller fixes may not always be listed explicitly but can often be addressed when related work is in progress.

As a valuable GitHub user, your feedback in this Discussion space, accompanied by appropriate tags and examples, plays a crucial role in shaping GitHub’s development. While we can’t guarantee immediate action on every piece of feedback, rest assured that we are attentive to your suggestions and always striving to enhance the GitHub experience.

Once again, thank you for taking the time to share your thoughts with us, even if the particular request is not currently in active development. Your input is valuable and contributes to the continuous improvement of GitHub!
